02-microarray/pathway-analysis_microarray_03_gsva.Rmd

```diff
@@ -405,11 +424,222 @@ Note that if we had duplicate gene identifiers here, we would not be able to set
+
+```{r}
+# Apply multiple testing correction and obtain stats
+stats_df <- topTable(contrasts_fit, number = nrow(expression_df)) %>%
+  tibble::rownames_to_column("Gene")
+```
+
+Let's take a peek at our results table.
+
+```{r}
+head(stats_df)
+```
+
+For each gene, each group's fold change in expression, compared to the average of the other groups is reported.
+
+By default, results are ordered from largest `F` value to the smallest, which means your most differentially expressed genes across all groups should be toward the top.
+
+This means `HALLMARK_UNFOLDED_PROTEIN_RESPONSE` appears to be the pathway that contains the most significant distribution of scores across subgroups. 
+
+## Visualizing Results
+
+Let's make a plot for our most significant pathway, `HALLMARK_UNFOLDED_PROTEIN_RESPONSE`.
+
+### Sina plot
+
+First we need to get our GSVA scores for this pathway in an appropriate format for `ggplot2`.
+
+```{r}
+annotated_results_df <-
+  # Pull out only the results for this pathway
+  gsva_results["HALLMARK_UNFOLDED_PROTEIN_RESPONSE", ] %>%
+  # Make this into a tidy data frame format
+  tibble::enframe("sample", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE") %>%
+  # Join the column with the group labels that we would like to plot and compare --
+  # Select the variables relevant to your data
+  dplyr::left_join(metadata %>% dplyr::select(
+    refinebio_accession_code,
+    subgroup
+  ),
+  # Tell the join what columns are equivalent and should be used as a key
+  by = c("sample" = "refinebio_accession_code")
+  )
+```
+
+Now let's make a sina plot so we can look at the differences between the `subgroup` groups using our GSVA scores.
+
+```{r}
+# Sina plot comparing GSVA scores for `HALLMARK_UNFOLDED_PROTEIN_RESPONSE` the `subgroup` groups in our dataset
+sina_plot <-
+  ggplot(
+    annotated_results_df, # Supply our annotated data frame
+    aes(
+      x = subgroup, # Replace with a grouping variable relevant to your data
+      y = HALLMARK_UNFOLDED_PROTEIN_RESPONSE, # Our pathway of interest
+      color = subgroup # Let's make the groups different colors too
+    )
+  ) +
+  # Add a boxplot that will have summary stats
+  geom_boxplot(outlier.shape = NA) +
+  # Make a sina plot that shows individual values
+  ggforce::geom_sina() +
+  # Adjust the plot background for better visualization
+  theme_bw()
+
+# Display plot
+sina_plot
+```
+
+Looks like the `Group 4` samples have lower GSVA scores for `HALLMARK_UNFOLDED_PROTEIN_RESPONSE` as compared to the `SHH` and `Group 3` scores.
+
+Let's save this plot to PNG.
+
+```{r}
+ggsave(file.path(plots_dir, "GSE37382_gsva_HALLMARK_UNFOLDED_PROTEIN_RESPONSE_sina_plot.png"),
+  plot = sina_plot
+)
+```
+
+## Write results to file
+
+Now let's write our all of our GSVA results to file.
+
+```{r}
+gsva_results %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column("pathway") %>%
+  readr::write_tsv(file.path(
+    results_dir,
+    "GSE37382_gsva_results.tsv"
+  ))
+```
```

Comment:
Wondering if we want to save the `topTable()` output as well? I don't have strong feelings about that. Or maybe I don't like that idea at all 🤔

Reply:
My thoughts on this were we probably want to stick to as close to GSVA as possible in this example, seeing as we have an existing differential expression example where the `topTable()` output is saved. Users in this particular situation will likely be more interested in the GSVA results, however, I can be convinced otherwise. 

I did not save the `topTable()` output in the last set of commits based on my thoughts above, but feel free to let me know if you feel more strongly about saving the output now.